### PR TITLE
Scatter plot wrapper for linear timeline views

### PR DESCRIPTION
**Pull request by smonero**
### Description

We wrap the `ScatterPlot` Recharts component for use in linear timeline views. This is more useful than the wrapper for linecharts for this specific use case.

Screen Shot 2022-03-09 at 8 39 46 PM

### Testing Performed

story,
local

### GitHub Issue

Fixes #

### CODE REVIEW

File: frontend/packages/core/src/Charts/linearTimeline.tsx [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+
+export interface LinearTimelineProps {
+  data: LinearTimelineData;
+  xAxisDataKey: string;
+  regularIntervalTicks?: boolean;
+  tickFormatterFunc?: (timestamp: number) => string;
+  legend?: boolean;
+}
+
+const LinearTimeline = ({
```
Comment (danielhochman) [inline frontend/packages/core/src/Charts/linearTimeline.tsx:41]: let's add some more documentation via comments.

File: frontend/packages/core/src/Charts/linearTimeline.tsx [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+
+export interface LinearTimelineData {
+  [lane: string]: LinearTimelineDataPoints;
+  // ...
+}
+
+export interface LinearTimelineProps {
+  data: LinearTimelineData;
+  xAxisDataKey: string;
+  regularIntervalTicks?: boolean;
```
Comment (danielhochman) [inline frontend/packages/core/src/Charts/linearTimeline.tsx:36]: document this prop, not obvious at first glance what the behavior difference is for true v false.

File: frontend/packages/core/src/stories/linearTimeline.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,47 @@
+import * as React from "react";
+import styled from "@emotion/styled";
+import type { Meta } from "@storybook/react";
+
+import type { LinearTimelineData } from "../Charts/linearTimeline";
+import LinearTimeline from "../Charts/linearTimeline";
+
+export default {
+  title: "Core/LinearTimeline",
```
Comment (danielhochman) [inline frontend/packages/core/src/stories/linearTimeline.stories.tsx:9]: can we nest all of the chart components under Core/Chart or similar?

File: frontend/packages/core/src/stories/linearTimeline.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  height: 200,
+});
+
+export const Primary = () => {
+  const mockData: LinearTimelineData = {
+    deploys: { points: [{ timestamp: 1588884888 }], shape: "cross", color: "purple" },
+    "k8s events": { points: [{ timestamp: 1588985888 }], shape: "triangle", color: "blue" },
+    explosions: {
+      points: [
+        { timestamp: 1588788888 },
```
Comment (danielhochman) [inline frontend/packages/core/src/stories/linearTimeline.stories.tsx:24]: is it possible to annotate individual events with tooltips? i looked at the recharts docs briefly for `Scatter` but it wasn't clear to me what the possibilities were for the `data` prop. also was reading https://stackoverflow.com/a/66084216

**Review by danielhochman — COMMENTED** [review]
looks really nice overall! good interface. a couple of questions and comments.

File: frontend/packages/core/src/stories/linearTimeline.stories.tsx (hunk 4 above)
Comment (smonero) [reply frontend/packages/core/src/stories/linearTimeline.stories.tsx:24]: yes it is. The payload object contains all the data for the point so with custom tooltip renders we can put whatever we want. The thing that seemed difficult was that there was no default styling, so if you don't pass any styling info the text is just black with no background or box. We can fix that by passing styling info.

File: frontend/packages/core/src/Charts/linearTimeline.tsx [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+      color: data[key].color,
+      laneID: index,
+    };
+  });
+  const formatIdsToNames = (value: string) => {
+    return <span>{mappingOfLaneIdsToNames[value]}</span>;
+  };
+
+  // TODO: Allow for proper styling and make things less hacky than "payload[0]"
+  const formatTooltip = ({ active, payload, label }) => {
```
Comment (jdslaugh) [inline frontend/packages/core/src/Charts/linearTimeline.tsx:95]: Do we have typings for this function?

File: frontend/packages/core/src/Charts/linearTimeline.tsx [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+} from "recharts";
+
+import { calculateDomainEdges, calculateTicks, localTimeFormatter } from "./helpers";
+
+export interface LinearTimelineDataPoint {
+  timestamp: number;
+  // more to come
+}
+
+// Note that shape can be a set of shapes denoted by preset strings ("circle", "square", etc.)
```
[the remark this answers is not in the document]
Comment (smonero) [reply frontend/packages/core/src/Charts/linearTimeline.tsx:20]: they do but they don't export the type so I guess we can't import it

[the remark this answers is not in the document]
Comment (smonero) [reply frontend/packages/core/src/Charts/linearTimeline.tsx:20]: the type is specifically `SymbolType | ReactElement<SVGElement> | ((props: any) => ReactElement<SVGElement>);`

[the remark this answers is not in the document]
Comment (smonero) [reply frontend/packages/core/src/Charts/linearTimeline.tsx:20]: I'll add a comment that talks about this

**Review by danielhochman — APPROVED** [review]
lgtm!

File: frontend/packages/core/src/Charts/linearTimeline.tsx (hunk 6 above)
[the remark this answers is not in the document]
Comment (jdslaugh) [reply frontend/packages/core/src/Charts/linearTimeline.tsx:19]: Alright, well we can probably copy it at least. Especially the `ReactElement<SVGElement> | ((props: any) => ReactElement<SVGElement>);` portion, that way it's somewhat known what shape should be

**Review by jdslaugh — APPROVED** [review]
Thanks!

**Pull request merged**